app/models/check_create.rb

```diff
@@ -0,0 +1,25 @@
+class CheckCreate
+  attr_reader :claim, :checked_by, :status
+
+  def initialize(claim_id:, checked_by:, status:)
+    @claim = Claim.find(claim_id)
+    @checked_by = checked_by
+    @status = status
+  end
+
+  def perform
+    return false unless claim.approvable?
+
+    create_check.valid? ? send_confirmation : false
```

Comment:
Is it possible that the claim to be marked as approved but `send_confirmation` may return false causing the user to see the message "Claim cannot be approved"? It feels like `perform` should return the result of the action and sending the email should be a side-effect